### PR TITLE
Parser: Fix boolean attribute explicit string value result

### PR DESCRIPTION
**Pull request by aduth**
Related: #7322

This pull request seeks to fix an issue in the implementation of #7322 where an explicit string value assigned as an attribute would be wrongly interpreted as `false` when assigned as a `boolean` attribute type. Both `<input disabled>` and `<input disabled="disabled">` should return `true` for a boolean attribute source type.

**Testing instructions:**

Ensure unit tests pass:

```
npm run test-unit blocks/api/parser.js
```

Repeat testing instructions from #7322.

### CODE REVIEW

File: blocks/api/parser.js [hunk 1]
```diff
@@ -19,6 +19,22 @@ import { isValidBlock } from './validation';
 import { getCommentDelimitedContent } from './serializer';
 import { attr, prop, html, text, query, node, children } from './matchers';
 
+/**
+ * Higher-order hpq matcher which enhances an attribute matcher to return true
```
Comment (tofumatt) [inline blocks/api/parser.js]: Nice docs 👍

File: blocks/api/parser.js [hunk 2]
```diff
@@ -19,6 +19,22 @@ import { isValidBlock } from './validation';
...
+ * or false depending on whether the original matcher returns undefined. This
+ * is useful for boolean attributes (e.g. disabled) whose attribute values may
+ * be technically falsey (empty string), though their mere presence should be
+ * enough to infer as a truthy value.
+ *
+ * @param {Function} matcher Original hpq matcher.
+ *
+ * @return {Function} Enhanced hpq matcher.``
```
Comment (tofumatt) [inline blocks/api/parser.js]: Not sure what the backticks at the end are for 🤷‍♂️

File: blocks/api/parser.js [hunk 3]
```diff
@@ -19,6 +19,22 @@ import { isValidBlock } from './validation';
...
+ */
+export const toBooleanAttributeMatcher = ( matcher ) => flow( [
+	matcher,
+	( value ) => value !== undefined,
```
Comment (tofumatt) [inline blocks/api/parser.js]: I almost asked "what about a `false` value?" but that should never happen. Not sure if it's worth adding docs just to remind the reader about that.

File: blocks/api/test/parser.js [hunk 4]
```diff
@@ -132,6 +174,19 @@ describe( 'block parser', () => {
 			expect( value ).toBe( true );
 		} );
 
+		it( 'should return the matcher\'s true boolean attribute value on explicit attribute value', () => {
```
Comment (tofumatt) [inline blocks/api/test/parser.js]: _This_ is definitely a case where I'm like #7366 for the win 😉

**Review by tofumatt — APPROVED** [review]
I dig it! A few comments/questions but approach is right on.

File: blocks/api/parser.js (hunk 2 above)
Comment (aduth) [reply blocks/api/parser.js]: >Not sure what the backticks at the end are for 🤷‍♂️

Fat fingered 😄

Comment (tofumatt) [reply blocks/api/parser.js]: That was my guess 😆

**Comment by aduth** [comment]
Thanks for the review @tofumatt . I pushed a few documentation updates in de1c2ea per your feedback.

**Comment by tofumatt** [comment]
Nice, I love the table 👍

**Comment by danielbachhuber** [comment]
Thanks @aduth

**Pull request merged**